tests/image_test.py

```diff
@@ -1224,6 +1225,26 @@ def test_load_img(self, tmpdir):
             loaded_im = image.load_img(filename_rgb, target_size=(25, 25),
                                        interpolation="unsupported")
 
+    def test_list_pictures(self, tmpdir):
+        os.mkdir(os.path.join(tmpdir, 'class1'))
+        os.mkdir(os.path.join(tmpdir, 'class2'))
+
+        filenames = ['test.png', 'test0.jpg', 'test-1.jpeg', '2test.bmp',
+                     '2-test.ppm', '3.png', '1.jpeg', 'test.bmp', 'test0.ppm']
+        subdirs = ['', 'subdir1', 'subdir2']
+        images = [tmpdir.mkdir(subdir).join(f) for subdir in subdirs
```

Comment:
This won't work. `tmpdir` will raise an exception if trying to create an existing dirextory. Your test can be this:
```
        filenames = ['test.png', 'test0.jpg', 'test-1.jpeg',
                     '2test.bmp', '2-test.ppm', '3.png']
        subdirs = [tmpdir.mkdir(s) for s in ['subdir1', 'subdir2']] + [tmpdir]
        images = [d.join(f).write('') for d in subdirs for f in filenames]

        found_images = image.list_pictures(tmpdir)
        assert len(found_images) == len(images)

        found_images = image.list_pictures(tmpdir, ext='png')
        assert len(found_images) == 6
```